contents/code/main.js

```diff
@@ -71,14 +72,14 @@ debug("");
 var block = false;
 
 // trigger debug output when client is activated
-workspace.clientActivated.connect(client => {
+workspace.windowActivated.connect(client => {
     if (!client) return;
     // debug(caption(client), geometry(client));
 });
 
 // trigger applying tile gaps when client is initially present or added
-workspace.clientList().forEach(client => onAdded(client));
-workspace.clientAdded.connect(onAdded);
+workspace.windowList().forEach(client => onAdded(client));
+workspace.windowAdded.connect(onAdded);
 
 function onAdded(client) {
```

Comment:
Signals don't send the client to the callback anymore, we need to figure out how to run onAdded without the client parameter.
As it is, new windows' listeners are unbound.